File: client/src/app/_models/repository.ts
// src/app/_models/repository.model.ts

export interface Repository {
    id: number;
    label: string;       // e.g., "JoVE"
    linkUrl: string;     // e.g., "https://www.jove.com/"
    imageUrl?: string;   // Cloudinary-secured image URL (may fallback to default)
    image?: File;        // Optional file for upload (sent via FormData)
}

File: API/Entities/Repository.cs
namespace API.Entities
{
    public class Repository
    {
        public int Id { get; set; }

        public string Label { get; set; } = string.Empty;

        public string LinkUrl { get; set; } = string.Empty;

        // ✅ Cloudinary-secured final URL for display
        public string? ImageUrl { get; set; }
    }
}
